src/parser/parser.rs

```diff
@@ -62,8 +62,8 @@ impl<'help, 'cmd> Parser<'help, 'cmd> {
         let mut parse_state = ParseState::ValuesDone;
         let mut pos_counter = 1;
 
-        // Already met any valid arg(then we shouldn't expect subcommands after it).
-        let mut valid_arg_found = false;
+        // Already met this valid arg(then we shouldn't expect subcommands after it).
+        let mut valid_arg_found = None;
```

Comment:
Turning this into an `Option` here gives us the nice option to tell the user which flag caused the subcommand to fail. Ideally they can then remove the listed subcommand and if another error pops up due to extra flags, they can repeat the process until all flags are removed.